python/ray/tests/test_client_builder.py

```diff
@@ -283,5 +284,98 @@ def test_address_resolution(call_ray_stop_only):
             del os.environ["RAY_ADDRESS"]
 
 
+def mock_connect(*args, **kwargs):
+    """
+    Force exit instead of actually attempting to connect
+    """
+    raise ConnectionError
+
+
+def has_client_deprecation_warn(warning, expected_replacement):
+    start = "Starting a connection through `ray.client` is deprecated"
+    message = str(warning.message)
+    if start not in message:
+        return False
+    if expected_replacement not in message:
+        return False
+    return True
+
+
+def test_client_deprecation_warn():
+    """
+    Tests that calling ray.client directly raises a deprecation warning with
+    a copy pasteable replacement for the client().connect() call converted
+    to ray.init style.
```

Comment:
Can you also add some tests for this with environment variables as well? (i.e. that they are not printed in the warnings)?